packages/vitest/src/node/core.ts

```diff
@@ -715,13 +729,12 @@ export class Vitest {
         }
       }
       finally {
-        // can be duplicate files if different projects are using the same file
-        const files = Array.from(new Set(specs.map(spec => spec.moduleId)))
-        const errors = this.state.getUnhandledErrors()
+        // TODO: wait for coverage only if `onFinished` is defined
```

Comment:
We do need it for `this.reportCoverage` below

Reply:
`this.reportCoverage` happens after `onFinished`, so I don't see anything that would stop this being a reality

I would go even further and move the printing logic to reporters

Reply:
I do wonder of course if it is a good idea to separate test run end from the coverage reporting. Playwright doesn't report coverage at all, for example